frontend-guide.md

```diff
@@ -0,0 +1,333 @@
+  newResponse.completedAt = newResponse.completedAt ? moment(newResponse.completedAt) : null
+  return newResponse
+}
+```
+
+To add more type safety around Backbone Model we add some boilerplate around the model creation.
+For each model we define the type of the return of the `get` function for each valid key. Calling `get` on a non valid `key` should throw a flow error, same for using the value as the incorrect type.
+To achieve this we have to define the type of the `Get` function as followed.
+From the same model as above
+
+```javascript
+type Get =
+  ((attr: 'id') => number) &
+  ((attr: 'name') => string) &
+  ((attr: 'createdAt') => moment) &
+  ((attr: 'completedAt') => ?moment) &
+  ((attr: 'studentIds') => Array<number>)
+```
+
+And attach the type of the `Get` function to the Backbone Model:
+
+```javascript
+export type MyModelT = Backbone.Model<Get> & typeof instanceMethods
+const MyModel: Class<Backbone.Model<Get>> = Backbone.Model.extend(instanceMethods)
+export default MyModel
+```
+
+More info about [Backbone])http://backbonejs.org/). You can also check the [annotated sources of Backbone](http://backbonejs.org/docs/backbone.html)
+
+
+### Backbone Router
+
+TODO
+
+### Flow
+
+[Flow](https://flow.org/) is a tool that add a layer of typing to javascript code. It relies on annotations in the code to figure out which type are the elements. When running `flow check`, flow will check the code base and throw all typing errors found. Flow contains the definition of the javascript API and can be completed by passing definition of libraries we use, like React, Backbone or Lodash. Other library definition can be in [flow-typed](https://github.com/flowtype/flow-typed).
+Importing an element to a file will also import its type but you can chose to only import its type.
+
+TODO Continue this section
+
+### React
+
+TODO [React](https://facebook.github.io/react/)
+
+### Mocha and Chai
+
+See Testing
+
+### 3rd Party libraries
+
+TODO
+
+- MathJax
+- TrackJS
+- Tracking and marketing
+  - Google Tag Manager
+  - Facebook Pixel Tracking
+  - Twitter (embedded)
+  - Youtube (embedded)
+
+
+## Best Practice and syntax
+
+
+### Javascript Style
+
+Refer to the [Javascript style document](https://github.com/frontrowed/guides/blob/master/javascript-style.md)
+
+### ESLint
+
+TODO
+
+## Testing
+
+We use [mocha](https://mochajs.org/) as the JS test framework and [chai](http://chaijs.com/) as the assertion library.
+
+We expect developers to add tests to logic and helper files. 
+
+All tests exist in the `test` folder of every application. We keep the same name and file path of the tested file for the test file:
+`js/helpers/my-nice-helper.js` should have a corresponding test file: `test/helpers/my-nice-helper-test.js`.
+
+By contention the first `describe` should contain the name of the tested file. Every nested `describe` should contain the name of the tested function.
+Example:
+```javascript
+describe("my-nice-helper", function() {
```

Comment:
I would point out why we use `function` instead of arrow functions here. If I recall, `mocha` does something sneaky with `this`